src/packs.c

```diff
@@ -52,19 +52,31 @@ static int download_pack(int oldversion, int newversion, char *module)
 		return 0;
 	}
 
-	string_or_die(&url, "%s/%i/pack-%s-from-%i.tar", content_url, newversion, module, oldversion);
-
-	err = swupd_curl_get_file(url, filename, NULL, NULL, true);
-	if (err) {
+	if (is_mix) {
+		string_or_die(&url, "%s/%i/pack-%s-from-%i.tar", MIX_STATE_DIR, newversion, module, oldversion);
+		err = link(url, filename);
+		printf("Linked %s to %s\n", url, filename);
```

Comment:
This should be printed after the `if (err) {...}` statement below, so it doesn't look like the link succeeded when it didn't.

Reply:
This is a valid concern; printing the progress message is not helpful when `link()` fails, so printing it after the error check below is better.

Reply:
Done.